From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Date: Wed, 13 Jul 2022 20:33:02 +0300
Subject: [PATCH 15/17] [FIO extra] boot: dont enable MD5,SHA1 if
 SIGNATURE_STRICT

Allowing these options to be disabled saves ~4k of SPL binary size,
that allows more SoC with smaller SPL requirements use the
SIGNATURE_STRICT setting.

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
(cherry picked from commit c36fbc4f921dbef49e537d6905ef9652c347f686)
---
 boot/Kconfig | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/boot/Kconfig b/boot/Kconfig
index d062df43816..c367b45f6d9 100644
--- a/boot/Kconfig
+++ b/boot/Kconfig
@@ -41,8 +41,11 @@ config BUTTON_CMD
 menuconfig FIT
 	bool "Flattened Image Tree (FIT)"
 	select HASH
-	select MD5
-	select SHA1
+	select MD5 if (!SPL_BUILD && !FIT_SIGNATURE_STRICT) || \
+		      (SPL_BUILD && !SPL_FIT_SIGNATURE_STRICT)
+	select SHA1 if (!SPL_BUILD && !FIT_SIGNATURE_STRICT) || \
+		       (SPL_BUILD && !SPL_FIT_SIGNATURE_STRICT)
+
 	imply SHA256
 	help
 	  This option allows you to boot the new uImage structure,
